aws-storage-s3/src/main/java/com/amplifyframework/storage/s3/operation/AWSS3StorageDownloadFileOperation.java

```diff
@@ -69,63 +69,49 @@ public AWSS3StorageDownloadFileOperation(
     public void start() {
         // Only start if it hasn't already been started
         if (transferObserver == null) {
-            String identityId;
+            String serviceKey = S3RequestUtils.getServiceKey(
+                    getRequest().getAccessLevel(),
+                    getRequest().getTargetIdentityId(),
+                    getRequest().getKey()
+            );
+
+            this.file = new File(getRequest().getLocal()); //TODO: Add error handling if path is invalid
 
             try {
-                identityId = AWSMobileClient.getInstance().getIdentityId();
+                transferObserver = storageService.downloadToFile(serviceKey, file);
+            } catch (Exception exception) {
```

Comment:
Can you `catch (AMoreSpecificException specificProblem)`?

Ref: [PMD's `AvoidCatchingGenericException` lint documentation](https://pmd.github.io/latest/pmd_rules_java_design.html#avoidcatchinggenericexception)

Reply:
I agree with this, but it is non-trivial to look for all possible ways that `TransferUtility` could throw a runtime exception due to its complexity. For example, it throws `IllegalArgumentException` for multiple reasons and it becomes kind of pointless to catch it and give a generic message for it (and repeat the process for other types of runtime exceptions it could potentially throw).